File: robocode2-server/src/main/java/net/robocode2/model/events/BotHitBotEvent.java
package net.robocode2.model.events;

import lombok.Value;
import net.robocode2.model.Point;

/**
 * Event sent when a bot collides with another bot.
 * 
 * @author Flemming N. Larsen
 */
@Value
public class BotHitBotEvent implements Event {

	/** Bot id of the bot hitting another bot */
	int botId;

	/** Bot id of the victim bot that got hit */
	int victimId;
	
	/** Energy level of the victim */
	double energy;
	
	/** Position of the victim */
	Point position;
	
	/** Flag specifying if the victim was rammed */
	boolean rammed;
}

File: robocode2-gui/src/main/kotlin/net/robocode2/gui/model/event/BotHitBotEvent.kt
package net.robocode2.gui.model.event

import net.robocode2.gui.model.comm.Content
import net.robocode2.gui.model.comm.ContentType

class BotHitBotEvent(
        val botId: Int,
        val victimId: Int,
        val energy: Double,
        val x: Double,
        val y: Double,
        val rammed: Boolean
) : Content(type = ContentType.BOT_HIT_BOT_EVENT.type)
